src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/cas2/TimelineEventsTransformer.kt

```diff
@@ -0,0 +1,57 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.cas2
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas2TimelineEvent
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.TimelineEventType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas2ApplicationEntity
+
+@Component("Cas2TimelineEventsTransformer")
+class TimelineEventsTransformer {
+  fun transformApplicationToTimelineEvents(jpa: Cas2ApplicationEntity): List<Cas2TimelineEvent> {
+    val timelineEvents: MutableList<Cas2TimelineEvent> = mutableListOf()
+
+    addSubmittedEvent(jpa, timelineEvents)
+
+    addStatusUpdateEvents(jpa, timelineEvents)
+
+    addNoteEvents(jpa, timelineEvents)
+
+    return timelineEvents.sortedByDescending { it.occurredAt }
+  }
+
+  private fun addNoteEvents(jpa: Cas2ApplicationEntity, timelineEvents: MutableList<Cas2TimelineEvent>) {
+    jpa.notes?.forEach {
+      timelineEvents += Cas2TimelineEvent(
+        type = TimelineEventType.cas2Note,
+        occurredAt = it.createdAt.toInstant(),
```

Comment:
hmm, good spot - I don't think `createdAt` should ever be null, because it's assigned when the entity is created (or should be!).
I think in the other examples:
* it's a mistake on the StatusUpdateEntity that createdAt is nullable
* the typing isn't clever enough to know that because we've checked `submittedAt` is not null, we don't need to guard against it in the `if` statement.

Reply:
> it's a mistake on the StatusUpdateEntity that createdAt is nullable

this is quite a big oversight so I'm fixing, as otherwise we can't sort them!